README.md

```diff
@@ -4,13 +4,14 @@
 name: Merge checks
 
 on:
-  pull_request:
-    types: [opened, synchronize]
+  push:
+    branches:
+      - feature/*
 
 jobs:
   merge-checks:
     runs-on: ubuntu-20.04
-
+    if: "!contains(github.event.head_commit.message, '[skip checks]')"
```

Comment:
Was passiert, wenn ich einen erfolgreichen CI-Run in einem PR habe, und dann einen Commit mit `skip checks` pushe? Bleibt der PR dann grün und ich kann ihn trotzdem mergen?

Reply:
Ja. In diesem Fall haben die merge-checks das Ergebnis `skipped`. Generell lässt das Ergebnis `skipped` neben `success` das mergen zu:

![Screenshot 2021-04-08 at 09 05 57](https://user-images.githubusercontent.com/610633/113983549-4f473280-984a-11eb-9720-37139999baae.png)

Reply:
D.h. obwohl "required" heißt "skipped" dann "success"? Das sollte denke ich nicht so sein.

Reply:
In der Tat scheint GitHub das so zu regeln. Uns kommt es an dieser Stelle zu gute. Ich finde das Ergebnis `skipped`, wenn wir den merge-check nicht wirklich ausführen, eigentlich sehr sauber. Oder würdest du das gerne anders regeln?

Reply:
Das ist doch blöd, oder? Ich will doch sicherstellen, ob die Checks erfolgreich waren bevor ich was merge. Egal ob im letzten Commit jetzt "skip ci" drinstand oder nicht. Zumindest sollte der Branch vor dem Merge immer komplett getestet sein, und das ist er ja nicht, wenn der letzte Stand "skipped" ist. Genau sowas soll ja verhindert werden.

Reply:
Wir möchten die Möglichkeit haben Checks zu überspringen, da mit https://github.com/moneymeets/moneymeets-pulumi/pull/88 dann Commits für alle Repositories gemacht werden. Wenn wir für jedes Repository dann zum Beispiel `lint-and-test` abwarten müssten, würde ein `pulumi up` Stunden dauern. Zugegeben sind die merge-checks im Vergleich zu `lint-and-test` schnell, trotzdem würde ich auch an dieser Stelle gerne `[skip checks]` einbauen.

Geht es dir um die `[skip checks]` speziell bei den merge-checks oder allgemein, auch zum Beispiel bei `lint-and-test`?

Reply:
Dann ist das nur wegen `pulumi up`? Warum geht pulumi da nicht direkt auf den master? Ob ich jetzt einen PR mit ausgeschaltetem Check mache und den sofort merge, oder ob ich dann direkt auf dem Zielbranch committe, macht doch effektiv keinen Unterschied.

Reply:
Ja, das Thema kam wegen `pulumi up` auf. Pulumi kann, sobald die Status Checks für master aktiviert sind, wie alle anderen auch, nicht mehr direkt auf master pushen. Um mergen zu können, müssen dann die angeforderten Status Checks vorliegen, wie etwa die `merge-checks`. Das Ergebnis darf nicht `failure` sein, aber `skipped` ud `success` sind ok. 

Wir können gerne heute Nachmittag auch noch einmal über die Hintergründe und wie das Thema geplant wurde sprechen. Ich habe es versucht so gut wie möglich im Epic und den Tickets zu dokumentieren, aber vielleicht ist das Thema dafür doch schon zu komplex geworden.